lib/slack.rb

```diff
@@ -1,11 +1,158 @@
 #!/usr/bin/env ruby
+require "pry"
+
+require_relative "channel"
+require_relative "user"
+require "httparty"
+require "dotenv"
+Dotenv.load
+
+class SlackError < StandardError; end
+
+class Slack
+  attr_reader :channels, :users
+
+  def initialize()
+    @channels = []
+    @users = []
+    get_lists
+  end
+
+  def get_lists()
+    query_parameters = {
+      token: ENV["SLACK_API_TOKEN"],
+      pretty: 1,
+    }
+    url = "https://slack.com/api/conversations.list?"
+    response = HTTParty.get(url, query: query_parameters)
+    response["channels"].each do |x|
+      @channels << Channel.new(x["name"], x["topic"]["value"], x["num_members"], x["id"])
+    end
+    url2 = "https://slack.com/api/users.list?"
+    response2 = HTTParty.get(url2, query: query_parameters)
+    response2["members"].each do |x|
+      @users << User.new(x["name"], x["id"], x["real_name"])
+    end
+  end
+
+  def lists_channels
+    puts "\nHere are the channels:"
+    @channels.each do |x|
+      puts "   - #{x.name}"
+    end
+    puts "---\n"
+  end
+
+  def lists_users
+    puts "\nHere are the users:"
+    @users.each do |x|
+      puts "   - #{x.user_name}"
+    end
+    puts "---\n"
+  end
+
+  def select_user(search)
+    user = @users.find { |x| x.user_name.downcase == search.downcase || x.user_id.downcase == search.downcase }
+  end
+
+  def select_channel(search)
+    channel = @channels.find { |x| x.name.downcase == search.downcase || x.slack_id.downcase == search.downcase }
+  end
+
+  def self.send_msg(recipient, text)
+    if recipient == nil
+      puts "Please select a channel or user first."
+      main
+    end
+
+    url = "https://slack.com/api/chat.postMessage"
+    if recipient.class == Channel
```

Comment:
The `if` and `elsif` here are identical, so you just need to check to see if `recipient` is not `nil`.